### DISCUSSION TITLE
How can i optimize the uploaded image only if it is not an svg file

### DISCUSSION THREAD

**Discussion by jydevgit** [joshembling/image-optimizer#11] [Q&A] [answered]
Actually, i like to have the image simply uploaded in case of svg and optimized in png in other cases.
How can i do this please?

**Comment by joshembling** [comment, marked as answer]
You could currently do this with FileUpload components that aren't set to multiple. E.g. if your component handles 1 image upload only, something like this will work:

```php
Forms\Components\FileUpload::make('image')
    ->image()
    ->optimize(function ($state) {
        foreach ($state as $image) {
            if (Str::endsWith($image->getFilename(), 'svg')) {
                return null;
            }
        }

        return 'png';
    }),
```

This won't work exactly how it is expected if you try it with the `multiple()` operator. However, if you have multiple chained and the image list doesn't contain svg's, this could work. 

I don't plan on adding or bettering this feature any time soon, however, if you'd like to try it and make a PR for me to look at you are welcome.

**Reply by jydevgit** [reply]
It works perfectly like that, thank you very much!
For the moment we are not using multiple() but if the case arises and forces us to find a solution, we will share it with you.
Thanks again.

**Reply by denizaygundev** [reply]
@joshembling thanks for sharing and for providing this amazing plugin, it's underrated. 

For anyone else that stumbles across this, I'm optimising to `webp` and ignoring unsupported image types that the GD driver doesn't support: 
```
->optimize(function ($state) {
    $skipFormats = ['tiff', 'heic', 'pdf'];

    return collect($state)->some(fn($image) =>
        Str::endsWith($image->getFilename(), $skipFormats)) ? null : 'webp';
    })
```
